### PR TITLE
Remove all instances of REQUIRED  and OPTIONAL

### PR DESCRIPTION
**Pull request by Paul-Clue**
see issue #868 

This PR removes all instances of REQUIRED and OPTIONAL. These changes already exist in presentational components. This PR updates the references of those priority names in the code itself to lessen any confusion in the future.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206594245402143

### CODE REVIEW

File: client/resources/aria-at-test-io-format.mjs
Comment (howard-e) [inline client/resources/aria-at-test-io-format.mjs:1645]: I don't think this has to be changed

File: client/resources/aria-at-test-io-format.mjs
Comment (howard-e) [inline client/resources/aria-at-test-io-format.mjs:1656]: Same with this, but if it were to change, the enum name may as well be changed too

**Review by howard-e — CHANGES_REQUESTED** [review]
@Paul-Clue thanks for taking this first step to update the repository to be consistent with the new language!

I think the key thing that may be missing here though is that a database migration is needed to update the older `TestPlanVersion.tests[].assertions[].priority` which still use REQUIRED and OPTIONAL as well.

File: server/util/convertAssertionPriority.js
Comment (alflennik) [inline server/util/convertAssertionPriority.js:8]: The TODO can be removed now.

File: client/resources/aria-at-test-io-format.mjs
Comment (alflennik) [reply client/resources/aria-at-test-io-format.mjs:1645]: Agreed. If you do change it, make sure to look at server/util/getMetrics.js which still uses the term "All Required." Probably better to just leave it as-is.

**Review by alflennik — CHANGES_REQUESTED** [review]
I concur with Howard that the main missing thing is the migration. Once that's done I'll run the changes and see if I can find any issues. As for the code, it looks good, not much to add to Howard's review.

File: server/util/convertAssertionPriority.js
Comment (Paul-Clue) [reply server/util/convertAssertionPriority.js]: Done

File: server/migrations/20240311192910-removeRequiredAndOptionalPriority.js
Comment (howard-e) [inline server/migrations/20240311192910-removeRequiredAndOptionalPriority.js:78]:
```suggestion
                            id = ?`,
```

File: shared/convertAssertionPriority.js
Comment (howard-e) [inline shared/convertAssertionPriority.js:6]:
```suggestion
 * @returns {null|string}
```

**Review by howard-e — CHANGES_REQUESTED** [review]
@Paul-Clue looks good! There's an issue I found with the migration when testing against the current prod data and have suggested an inline fix.

Also, you can remove `REQUIRED` and `OPTIONAL` for the `AssertionPriority` enum in graphql-schema.js

**Comment by alflennik** [comment]
@Paul-Clue looks like the tests are failing, any idea about that?

File: shared/getMetrics.js
Comment (howard-e) [inline shared/getMetrics.js:194]: So with this change, because https://github.com/w3c/aria-at-app/blob/3808ed7f46642564718978aabd2765dd96c748c3/shared/getMetrics.js#L46 uses the priority string in combination with what has been named through the client's various `queries.js` files, it will incorrectly display the expected values for anywhere `TestPlanResultsTable` is getting used.

ie. [`queries.js`](https://github.com/w3c/aria-at-app/blob/3808ed7f46642564718978aabd2765dd96c748c3/client/components/TestRun/queries.js#L39) provides `requiredAssertionResults` but this change to `getMetrics` is expecting `mustRequiredAssertionResults`.

The following screenshot also shows that for each command, there are 2+ assertions but the command's assertion level status only shows `2 passed`, which would then also affect the total calculation for the entire test:
screenshot showing inaccurately calculated numbers

2 things I think could happen here:
1. Revert the change to `getMetrics` until updating each of the named `(required|optional)AssertionResults` instances. But this may cause some confusion until then.
2. Update the named instances of `(required|optional)AssertionResults` as appropriate which means renaming the value being exported from `getMetrics` and also adding a migration to update metrics across all the TestPlanReports

**Review by howard-e — CHANGES_REQUESTED** [review]

File: shared/getMetrics.js
Comment (Paul-Clue) [reply shared/getMetrics.js:194]: Hey @howard-e all of the changes have been made and a migration created.

File: shared/getMetrics.js
Comment (alflennik) [inline shared/getMetrics.js:223]: I think `requiredFormatted` needs to be `mustFormatted` and `optionalFormatted` needs to be `shouldFormatted`, right?

**Comment by alflennik** [comment]
@Paul-Clue fantastic work, I just have one comment. I verified everything else is working and once you address that I'll approve.

File: shared/getMetrics.js
Comment (Paul-Clue) [reply shared/getMetrics.js]: Ok done!

**Review by alflennik — APPROVED** [review]
Thanks for your considerable persistence chasing down all the details needed to get this task done! 😜

File: client/components/CandidateReview/CandidateTestPlanRun/queries.js
Comment (howard-e) [inline client/components/CandidateReview/CandidateTestPlanRun/queries.js:124]:
```suggestion
                    shouldAssertionResults: assertionResults(priority: SHOULD) {
```

**Review by howard-e — COMMENTED** [review]
Looks good to go after my last inline comment

File: shared/getMetrics.js
Comment (howard-e) [reply shared/getMetrics.js:194]: Great! Thanks for going the route of #2 as well. Definitely makes things cleaner for the future!

File: client/components/CandidateReview/CandidateTestPlanRun/queries.js
Comment (Paul-Clue) [reply client/components/CandidateReview/CandidateTestPlanRun/queries.js]: Got it! Thanks @howard-e !

**Review by howard-e — APPROVED** [review]
Looks good! Thanks for the continued changes to perfect and support this change @Paul-Clue!

**Comment by ccanash** [comment]
See issue #926

**Pull request merged**
